File: 713-subarray-product-less-than-k/subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k <= 1:
        
            return 0

        product = 1
        result = 0

        left = 0
        right = 0

        while right < len(nums):
            product *= nums[right]
            while product >= k:
                product /= nums[left]
                left += 1
            result += right - left + 1
            right += 1

        return result

        

File: 713-subarray-product-less-than-k/subarray-product-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numSubarrayProductLessThanK = function(nums, k) {
    let left = 0 
    let curr = 1
    let answer = 0

    if (k <= 1){
        return 0
    }
    for (let right = 0; right < nums.length; right++){
        curr *= nums[right]
        while(curr >= k){
            curr /= nums[left]
            left++
        }
        answer += right - left + 1
    }
    return answer
    
};